spring-resttemplate-2/src/main/java/com/baeldung/resttemplate/json/consumer/controller/UserConsumerController.java

```diff
@@ -0,0 +1,49 @@
+package com.baeldung.resttemplate.json.consumer.controller;
+
+import com.baeldung.resttemplate.json.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+
+@RestController
+public class UserConsumerController {
+
+    @Autowired
+    private final RestTemplate restTemplate;
+
+    public UserConsumerController(RestTemplate restTemplate) {
```

Comment:
You don't need to both provide constructor injection and field injection. You may find that you can take `@Autowired` off the field and things work. Depending on the version of Spring you may need to add `@Autowired` to the constructor, though these days it's not needed.